Formula/h/hdf5.rb

```diff
@@ -43,32 +42,20 @@ def install
               "settingsdir=$(libdir)",
               "settingsdir=#{pkgshare}"
 
-    system "autoreconf", "--force", "--install", "--verbose"
-
-    args = %W[
-      --disable-dependency-tracking
-      --disable-silent-rules
-      --enable-build-mode=production
-      --enable-fortran
-      --enable-cxx
-      --prefix=#{prefix}
-      --with-szlib=#{Formula["libaec"].opt_prefix}
-    ]
-    args << "--with-zlib=#{Formula["zlib"].opt_prefix}" if OS.linux?
-
-    system "./configure", *args
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
```

Comment:
To enable C++ and Fortran:

```suggestion
    system "cmake", "-S", ".", "-B", "build", "-DHDF5_BUILD_FORTRAN:BOOL=ON", "-DHDF5_BUILD_CPP_LIB:BOOL=ON", *std_cmake_args
```